src/components/EditPasswordForm/EditPasswordForm.tsx

```diff
@@ -26,35 +24,36 @@ type Props = {
   submitting: boolean;
   showOldPasswordField?: boolean;
   showResetTokenField?: boolean;
+  email?: string;
+  onResendEmailClick?: () => void;
 };
 
-const EditPasswordForm: React.FC<Props> = ({ onSubmit, onChange, onBlur, showOldPasswordField, showResetTokenField, value, errors, submitting }: Props) => {
+const EditPasswordForm: React.FC<Props> = ({
+  onSubmit,
+  onChange,
+  onBlur,
+  showOldPasswordField,
+  showResetTokenField,
+  value,
+  errors,
+  submitting,
+  email,
+  onResendEmailClick,
+}: Props) => {
   const { t } = useTranslation(['account', 'user']);
-  const email = useAccountStore((state) => state.user?.email);
-  const [errorMessage, setErrorMessage] = useState<string>();
-  const feedbackMessage = errors.form || errorMessage;
-  const resendEmailClickHandler = async () => {
-    try {
-      if (email) {
-        await resetPassword(email, '');
-      }
-    } catch (error: unknown) {
-      setErrorMessage(t('user:account.resend_mail_error'));
-    }
-  };
   return (
     <form onSubmit={onSubmit} data-testid={testId('forgot-password-form')} noValidate className={styles.forgotPasswordForm}>
-      {feedbackMessage && <FormFeedback variant="error">{feedbackMessage}</FormFeedback>}
-      <h2 className={styles.title}>{!showOldPasswordField || !showResetTokenField ? t('reset.password_reset') : t('user:account.add_password')}</h2>
-      {!showOldPasswordField || !showResetTokenField ? null : (
+      {errors.form && <FormFeedback variant="error">{errors.form}</FormFeedback>}
+      <h2 className={styles.title}>{showOldPasswordField && showResetTokenField ? t('user:account.add_password') : t('reset.password_reset')}</h2>
+      {showOldPasswordField && showResetTokenField && (
```

Comment:
We should write this down for later, but I still have trouble understanding why this logic is in place. The edit password form implements two methods:

- Using your old password
- Using a password reset token received via mail

We assume a mix of these methods is possible, while that should not be the case. This can be solved by using a single prop, for example; `method: 'token' | 'old-password'`.

I assume that `showOldPasswordField` and `showResetTokenField` should never be `true` simultaneously. Is that correct? If so, I'm afraid the condition is still wrong.

Reply:
As we discussed, the logic is correct but a bit weird, perhaps due to using existing props/variables for new use-cases.

We'll look into this later.